packages/SettingsProvider/src/com/android/providers/settings/SettingsState.java

```diff
@@ -1153,6 +1153,9 @@ private void parseSettingsLocked(TypedXmlPullParser parser)
                     fromSystem = parser.getAttributeBoolean(null, ATTR_DEFAULT_SYS_SET, false);
                     tag = getValueAttribute(parser, ATTR_TAG, ATTR_TAG_BASE64);
                 }
+                if (isSettingBlockedFromReadAndWrite(name, mKey)) {
```

Comment:
This can be moved up to line 1144 (after `name` is set) to avoid wasting time obtaining other attributes.